File: KuantumKaos/java/kuantumKaosJava/src/KaranlikMadde.java
public class KaranlikMadde extends KuantumNesnesi implements IKritik {

    @Override
    public void analizEt() {
        stabilite -= 20;
        System.out.println("Karanlık madde dalgalanıyor!");

        if (stabilite <= 0) {
            throw new KuantumCokusuException(id);
        }
    }

    @Override
    public void acilDurumSogutmasi() {
        stabilite += 30;
        System.out.println("Karanlık madde soğutuldu.");
    }
}

File: KuantumKaos/javascript/js_kaos/KaranlikMadde.js
import { KuantumNesnesi } from "./KuantumNesnesi.js";
import { IKritik } from "./IKritik.js";
import { KuantumCokusuException } from "./KuantumCokusuException.js";

export class KaranlikMadde extends KuantumNesnesi {
    analizEt() {
        this.stabilite -= 20;
        console.log("Karanlık madde dalgalanıyor!");

        if (this.stabilite <= 0) {
            throw new KuantumCokusuException(this.id);
        }
    }

    acilDurumSogutmasi() {
        this.stabilite += 30;
        console.log("Karanlık madde soğutuldu.");
    }
}
